packages/manager/apps/catalog-revamp/src/hooks/useCatalog.ts

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from 'react';
+import { groupBy, includes, isEmpty, map, uniq } from 'lodash-es';
+
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+export const useCatalog = (catalogData: Product[] = []) => {
+  const [products, setProducts] = useState(catalogData);
+  const [selectedCategories, setSelectedCategories] = useState([]);
+  const [selectedUniverses, setSelectedUniverses] = useState([]);
+  const [searchText, setSearchText] = useState('');
+  const [items, setItems] = useState(<any>[]);
+  const [categories, setCategories] = useState([]);
+  const [universes, setUniverses] = useState([]);
+
+  const groupItems = (itemsToGroup: any) => {
```

Comment:
you can move all of these functions in a typescript utils files

groupItems()
filterByEnum()
getAvailableCategories()
matchCategories()
matchUniverses()
matchSearchText()
filterItems()
getUniverses()

The goal is to keep the typescript file if we switch framework, and to test make unit test on theses functions more easy

Reply:
Done ✅